src/lib.rs

```diff
@@ -919,11 +953,48 @@ unsafe extern "C" fn fatal_handler(_: *mut os::raw::c_void, msg_raw: *const os::
     panic!("Duktape fatal error: {}", msg)
 }
 
+pub struct StackRAII {
+    ctx: *mut duk_sys::duk_context,
+    idx: i32,
+}
+impl StackRAII {
+    pub fn new(ctx: *mut duk_sys::duk_context) -> Self {
```

Comment:
This needs to be unsafe or something. Or I could cause UB with just `drop(StackRAII::new(std::ptr::null_mut()))`